corehq/apps/sms/api.py

```diff
@@ -241,19 +243,27 @@ def process_sms_registration(msg):
     keyword1 = text_words[0] if len(text_words) > 0 else ""
     keyword2 = text_words[1].lower() if len(text_words) > 1 else ""
     keyword3 = text_words[2] if len(text_words) > 2 else ""
+    keyword4 = text_words[3] if len(text_words) > 3 else ""
+    keyword5 = text_words[4] if len(text_words) > 4 else ""
+    cleaned_phone_number = strip_plus(msg.phone_number)
     if keyword1 in REGISTRATION_KEYWORDS and keyword2 != "":
         domain = Domain.get_by_name(keyword2, strict=True)
         if domain is not None:
             if keyword3 in REGISTRATION_MOBILE_WORKER_KEYWORDS and domain.sms_mobile_worker_registration_enabled:
-                #TODO: Register a PendingMobileWorker object that must be approved by a domain admin
-                pass
+                if keyword4 != '' and keyword5 != '':
+                    raise UndefinedUsernameException()
+                new_user = CommCareUser.create(domain.name, keyword4, keyword5)
+                new_user.add_phone_number(cleaned_phone_number)
+                new_user.save_verified_number(domain.name, cleaned_phone_number, True, None)
```

Comment:
@czue @TylerSheffels We shouldn't have to worry about the phone number being registered in the system because the only way we'd get here is if the phone number already doesn't exist (and we always process inbound sms for a single phone number serially so a race condition can't happen either).